### PR TITLE
search's source value is lost when False or throws exception when list during clone

### PR DESCRIPTION
**Pull request by mumino**

### CODE REVIEW

File: elasticsearch_dsl/search.py [hunk 1]
```diff
@@ -294,7 +295,10 @@ def _clone(self):
 
         s._response_class = self._response_class
         s._sort = self._sort[:]
-        s._source = self._source.copy() if self._source else None
+        if hasattr(self._source, "copy"):
```
Comment (HonzaKral) [inline elasticsearch_dsl/search.py]: actually since `copy.copy` also works on `dict`s we don't need this `if`.

Comment (mumino) [inline elasticsearch_dsl/search.py]: I left it if something needs special care. but I don't know any special case atm.

**Comment by HonzaKral** [comment]
Thank you very much for this fix!

**Comment by levylll** [comment]
I still get this error....
I install my elasticsearch_dsl with pip.
😢

**Comment by levylll** [comment]
@HonzaKral Please upload the new package to pypi...

**Pull request merged**